src/lib.rs

```diff
@@ -855,9 +855,8 @@ impl<T> Async<T> {
     /// inner.set_nonblocking(false)?;
     /// # std::io::Result::Ok(()) });
     /// ```
-    pub fn into_inner(mut self) -> io::Result<T> {
-        let io = self.io.take().unwrap();
-        Reactor::get().remove_io(&self.source)?;
+    pub fn into_inner(self) -> io::Result<T> {
+        let Self { source: _, io } = self;
```

Comment:
I just realized, this no longer returns an error if `remove_io()` fails. Can you fix it so that the error isn't ignored here?

Reply:
Done.